irc/nickserv.go

```diff
@@ -1041,28 +1162,16 @@ func nsPasswdHandler(service *ircService, server *Server, client *Client, comman
 		}
 	case 3:
 		target = client.Account()
+		newPassword = params[1]
 		if target == "" {
 			errorMessage = `You're not logged into an account`
-		} else if params[1] != params[2] {
+		} else if newPassword != params[2] {
 			errorMessage = `Passwords do not match`
 		} else {
 			if !nsLoginThrottleCheck(service, client, rb) {
 				return
 			}
-			accountData, err := server.accounts.LoadAccount(target)
-			if err != nil {
-				errorMessage = `You're not logged into an account`
-			} else {
-				hash := accountData.Credentials.PassphraseHash
-				if hash != nil && passwd.CompareHashAndPassword(hash, []byte(params[0])) != nil {
-					errorMessage = `Password incorrect`
-				} else {
-					newPassword = params[1]
-					if newPassword == "*" {
-						newPassword = ""
-					}
-				}
-			}
+			errorMessage = nsConfirmPassword(server, target, params[0])
```

Comment:
The old logic set newPassword to "" when it was "*" (certfp-only). I think that's still needed, no?

Reply:
Oh, wow, yes, this is really bad

Reply:
Actually this is not quite as bad as I thought it was, there was a defense against it:

https://github.com/ergochat/ergo/blob/9c77f89bc22410221cf9cb4146d9a77735c16919/irc/accounts.go#L504-L506